tests/dragonfly/replication_test.py

```diff
@@ -3114,11 +3114,21 @@ async def test_partial_replication_on_same_source_master(df_factory, use_takeove
     if use_takeover:
         # Promote first replica to master
         await c_replica1.execute_command(f"REPLTAKEOVER 5")
+        if backlog_len > 1:
+            await c_replica1.execute_command("SET bar foo")
+            await c_replica1.execute_command("SET foo bar")
+
     else:
         # Promote first replica to master
         await c_replica1.execute_command(f"REPLICAOF NO ONE")
-        # Send 2 more commands to be propagated to second replica
-        # Sending 2 more commands will result in partial sync if allow_partial_sync_with_lsn_diff is equal or higher
+        # Not sure what was the point of this test/flow. To assert that we have a bug ?
```

Comment:
`if use_takeover or (backlog_len > 0 and not use_takeover):` 

I still don't understand this, c2 contains changes that c1 does not have and we manage to make c2 a replica of c1 and partially sync ?! This flow is broken. If we can't guarantee consistency why would we allow partial sync ? For that flow data(c2) != data(c1)